EIPS/eip-2935.md

```diff
@@ -52,25 +53,94 @@ def process_block_hash_history(block: Block, state: State):
             state.insert_slot(HISTORY_STORAGE_ADDRESS, ancestor.number % HISTORY_SERVE_WINDOW, ancestor.hash)
 ```
 
-Note that if this is the fork block, then it persists the additional requisite history that could be needed while resolving `BLOCKHASH` opcode for all of the `HISTORY_SERVE_WINDOW` ancestors (up until genesis).
+Note that if this is the fork block, then it persists the additional requisite history that could be needed while resolving `BLOCKHASH` opcode for all of the `HISTORY_SERVE_WINDOW` > `BLOCKHASH_OLD_WINDOW` ancestors (up until genesis).
 
 For resolving the `BLOCKHASH` opcode this fork onwards (`block.timestamp >= FORK_TIMESTAMP`), switch the logic to:
 
 ```python
 def resolve_blockhash(block: Block, state: State, arg: uint64):
   # check the wrap around range
-  if arg >= block.number or arg < max(block.number - HISTORY_SERVE_WINDOW, 0)
+  if arg >= block.number or (arg + HISTORY_SERVE_WINDOW) < block.number
     return 0
 
   return state.load_slot(HISTORY_STORAGE_ADDRESS, arg % HISTORY_SERVE_WINDOW)
 ```
 
+Exact evm assembly that can be used for the blockhash contract:
+
+```
+// check for <=32 (<33) byte size or else revert
+calldatasize
+push1 0x21
+gt
+push1 0x0a
+// jump and skip revert
+jumpi
```

Comment:
do we want to make the happy path more expensive? I would rather jump at the end if an error occurs. That's if the total bytecode is less than 256 bytes of course. Which it currently is.

Reply:
was thinking about the same, but this is the most serial code style for the manners of analysis and understanding, but yea can update to make the happy path without jumps

Reply:
updated